source/core/src/main/com/csse3200/game/entities/factories/WeaponFactory.java

```diff
@@ -25,150 +13,102 @@
  */
 public class WeaponFactory extends LoadedFactory {
 
-    private static final Logger logger = LoggerFactory.getLogger(PlayerFactory.class);
+    private static final Logger logger = LoggerFactory.getLogger(WeaponFactory.class);
 
-    /**
-     * Create a collectible melee weapon from specification
-     *
-     * @param specification the specification of the weapon.
-     * @return MeleeWeapon the newly created collectible of type melee weapon.
-     * @throws IllegalArgumentException if the specification is invalid.
-     */
     private MeleeWeapon createMelee(String specification) throws IllegalArgumentException {
-        return switch (specification) {
-            case "knife" -> new Knife();
-            case "pickaxe" -> new Pickaxe();
-            default -> throw new IllegalArgumentException("Invalid melee weapon specification: " + specification);
+        WeaponConfig.WeaponData weaponData = WeaponConfig.getWeaponData(specification);
+        if (weaponData == null) {
+            throw new IllegalArgumentException("Invalid melee weapon specification: " + specification);
+        }
+
+        MeleeWeapon meleeWeapon = new MeleeWeapon() {
```

Comment:
Instead of overwritting MeleeWeapon on instantiation, you can just create a new class (ex: ConcreateMeleeWeapon) within the same package as MeleeWeapon, and use the output dictionary of getWeaponData to create it accordingly.